notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb

```diff
@@ -0,0 +1,1557 @@
+    "    vision_data, text_data = collect_vision_text_data(dataloader, opt_init_steps)\n",
+    "    return vision_data, text_data"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "3c0f6587",
+   "metadata": {},
+   "source": [
+    "The [ESC-50](https://github.com/karolpiczak/ESC-50) dataset is used to quantize the audio modality of the ImageBind model.\n",
+    "Dataset is a labeled collection of 2000 environmental audio recordings suitable for benchmarking methods of environmental sound classification.\n",
+    "The dataset consists of 5-second-long recordings organized into 50 semantical classes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "2a2d4357",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import torchaudio\n",
+    "\n",
+    "def collect_audio_data(dataloader, init_steps=300):\n",
+    "    audio_data = []\n",
+    "    for _, batch in zip(range(init_steps), itertools.islice(dataloader, 0, init_steps)):\n",
+    "        with torch.no_grad():\n",
+    "            audio_data.append(batch)\n",
+    "    return audio_data\n",
+    "\n",
+    "def prepare_audio_dataset():\n",
+    "    audio_dataset = load_dataset(\"ashraq/esc50\", \"clean\", streaming=True)\n",
+    "    train_dataset = audio_dataset[\"train\"].shuffle(seed=42, buffer_size=1000)\n",
+    "\n",
+    "    def collate_fn(examples):\n",
+    "        assert len(examples) == 1\n",
+    "        with tempfile.TemporaryDirectory() as tempdir:\n",
+    "            f_name = os.path.join(tempdir, 'audio.wav')\n",
+    "            audio_data = torch.from_numpy(examples[0]['audio']['array'].astype(np.float32)).unsqueeze(0)\n",
+    "            torchaudio.save(f_name, audio_data, examples[0]['audio']['sampling_rate'])\n",
+    "            return data.load_and_transform_audio_data([f_name], \"cpu\")\n",
+    "\n",
+    "    dataloader = torch.utils.data.DataLoader(train_dataset, collate_fn=collate_fn, batch_size=1)\n",
+    "    calibration_data = collect_audio_data(dataloader)\n",
+    "    return calibration_data"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "768c5f5f",
+   "metadata": {},
+   "source": [
+    "Create a quantized model from the pre-trained `FP16` model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4137cd4b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "vision_data, text_data = prepare_vision_text_dataset()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1d6e9780",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import nncf\n",
+    "from openvino.runtime import Core\n",
+    "from openvino.runtime import serialize\n",
+    "\n",
+    "def quantize_openvino_model(modality, calibration_data):\n",
+    "    model = Core().read_model(fp_model_paths[modality])\n",
+    "    quantized_model = nncf.quantize(\n",
```

Comment:
NNCF logs are too long and even confusing (ignoring only reduce layer, but many more layers are ignored due to Transformer model type). I suggest to disable "debug" at least.

Not a stopper for this notebook but we should decide what to do in general.